Loop/Managers/LoopDataManager.swift

```diff
@@ -193,36 +197,49 @@ final class LoopDataManager: LoopSettingsAlerterDelegate {
     private var lockedSettings: Locked<LoopSettings>
 
     var settings: LoopSettings {
-        get {
-            lockedSettings.value
+        lockedSettings.value
+    }
+
+    @discardableResult func mutateSettings(_ changes: (_ settings: inout LoopSettings) -> Void) -> LoopSettings {
+        setSettingsWithResult({ (settings) -> LoopSettings in
+            changes(&settings)
+            return settings
+        })
+    }
+
+    private func setSettingsWithResult(_ changes: (_ settings: inout LoopSettings) -> LoopSettings) -> LoopSettings {
```

Comment:
I think this pattern that was copied from elsewhere can be cleaned up. We're only exposing one new method `mutateSettings`, which can just call lockedSettings.mutate directly without needing a helper function that adds functionality to return the modified structure, since the return value is discarded.

Reply:
Yes, I was wondering about this as well. Will update.